app/controllers/scheduler_controller.rb

```diff
@@ -30,14 +31,28 @@ def destroy
   private
 
   def set_enrollment
-    @enrollment = Enrollment.where(user: current_user, course_instance_id: params[:course_instance_id]).first_or_create
+    course_instance_id =
+      if params[:course_instance_id].present?
+        params[:course_instance_id]
+      else
+        CourseInstance.search(params[:course_title], current_user, @search_date || Date.today)
+                      .first
+                      .try(:id)
+      end
+    return false if course_instance_id.nil?
+
+    @enrollment = Enrollment.where(user: current_user, course_instance_id: course_instance_id)
+                            .first_or_create
   end
 
   # Set the course instances to be rendered in the schedule
   # This is for the JSON feed that fullcalendar requires
   def set_scheduled_meetings
-    # timecop should be set here to get the correct enrollments in the query
-    @scheduled_meetings = current_user.enrolled_courses.ongoing(@search_date || Date.today).includes(:meetings, :course).flat_map(&:meetings).flat_map(&:scheduled_meetings)
+    @scheduled_meetings = current_user.enrolled_course
+                                      .ongoing(@search_date || Date.today)
+                                      .includes(:meetings, :course)
+                                      .flat_map(&:meetings)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.